scripts/run_soupX.R

```diff
@@ -45,20 +46,134 @@ clusters_file <- opts$cluster
+umap <- seurat_obj@reductions$umap@cell.embeddings
+sc  <- setClusters(sc, setNames(meta$seurat_clusters, rownames(meta)))
+sc  <- setDR(sc, umap)
+head(meta)
+
+######################################################################################
+######################################################################################
+# With defined clusters, run the main SoupX function, calculating ambient RNA profile.
+######################################################################################
+######################################################################################
+# to set the contamination fraction to 20% for all cells.
+# we shouldn't remove more than 20% background
+# sc = setContaminationFraction(sc, 0.2)
+
+# Estimate contamination fraction rho
+# The posterior distribution is calculated using a Poisson likelihood with a gamma distribution prior, parametrised by its mean priorRho and standard deviation priorRhoStdDev. The dotted line in the above plot shows the prior distribution. The default parameters have been calibrated to be fairly non-specific 
+# with a slight preference towards values of rho in the 0% to 10% range which is most commonly seen for fresh (i.e. not nuclear) single cell experiments.
+# The default values place only a very weak constraint, as can be seen by setting a uniform prior.
+sc <- autoEstCont(sc) #priorRhoStdDev = 0.3
+print(sc)
+
+# Genes with highest expression in background. These are often enriched for ribosomal proteins.
+head(sc$metaData)
+head(sc$soupProfile[order(sc$soupProfile$est, decreasing = TRUE), ], n = 20)
+
+# To visualize the distribution of expression (relative to what would be expected were each cell pure background) across all cells in the data set. 
+# When no geneset is provided, the function will try and guess which genes might be useful.
+print(plotMarkerDistribution(sc))
+
+# Correcting expression profile
+# This will contain a corrected matrix to be used in place of the original table of counts in downstream analyses.
+# use roundToInt option to make sure we output integer matrix.
+out <- adjustCounts(sc, roundToInt = TRUE) 
+
+# Optional but we can keep original matrix
+# Attention as if we do, then for the next step we need to ensure we are using the correct assay, i.e., `RNA_SoupX`
+seurat_obj[["RNA_SoupX"]] <- CreateAssayObject(counts = seurat_obj@assays$RNA@counts)
+seurat_obj@assays$RNA@counts <- out
+
+# Setup assay for seurat object
+DefaultAssay(seurat_obj) <- "RNA_SoupX"
+seurat_obj
+
+
+#########################################################################################
+# Investigating changes in expression
+# Before proceeding let's have a look at what this has done. 
+# We can get a sense for what has been the most strongly decreased 
+# by looking at the fraction of cells that were non-zero now set to zero after correction.
+# Rename to toc or it gives NULL
+toc = filtered_matrix
+toc = sc$toc
+
+cntSoggy = rowSums(sc$toc > 0)
```

Comment:
Count "soggy" versus "strained"? Is this a cute way to say unfiltered vs filtered, something like that?

Reply:
This comparison between `filtered vs adjusted matrix`, i.e., after correction (not raw matrix).